trin-history/src/jsonrpc.rs

```diff
@@ -35,236 +37,262 @@ impl HistoryRequestHandler {
                         }
-                        Err(msg) => Err(format!("Invalid SendOffer params: {:?}", msg)),
                     };
                     let _ = request.resp.send(response);
                 }
-                HistoryEndpoint::Ping => {
-                    let response = match PingParams::try_from(request.params) {
-                        Ok(val) => match self.network.overlay.send_ping(val.enr.into()).await {
-                            Ok(pong) => Ok(pong.into()),
-                            Err(msg) => Err(format!("Ping request timeout: {:?}", msg)),
-                        },
-                        Err(msg) => Err(format!("Invalid Ping params: {:?}", msg)),
+                HistoryEndpoint::Ping(enr, _) => {
+                    let enr = convert_enr(enr);
+                    let response = match self.network.overlay.send_ping(enr).await {
+                        Ok(pong) => Ok(json!({
+                            "enrSeq": pong.enr_seq,
+                            "dataRadius": *self.network.overlay.data_radius()
+                        })),
+                        Err(msg) => Err(format!("Ping request timeout: {:?}", msg)),
                     };
+
                     let _ = request.resp.send(response);
                 }
                 HistoryEndpoint::RoutingTableInfo => {
-                    let bucket_entries_json =
-                        bucket_entries_to_json(self.network.overlay.bucket_entries());
+                    let response = Ok(bucket_entries_to_json(
+                        self.network.overlay.bucket_entries(),
+                    ));
 
-                    let _ = request.resp.send(Ok(bucket_entries_json));
+                    let _ = request.resp.send(response);
                 }
             }
         }
     }
+
+    async fn recursive_find_content(
+        &mut self,
+        content_key: HistoryContentKey,
+        is_trace: bool,
+    ) -> Result<Value, String> {
+        // Check whether we have the data locally.
+        let local_content: Option<Vec<u8>> =
+            match self.network.overlay.store.read().get(&content_key) {
+                Ok(Some(data)) => Some(data),
+                Ok(None) => None,
+                Err(err) => {
+                    error!(
+                        error = %err,
+                        content.key = %content_key,
+                        "Error checking data store for content",
+                    );
+                    None
+                }
+            };
+        let (content, closest_nodes) = match local_content {
+            Some(val) => (Some(val), vec![]),
+            None => {
+                self.network
+                    .overlay
+                    .lookup_content(content_key.clone())
+                    .await
+            }
+        };
+        let content = content.unwrap_or_default();
+        let response = match is_trace {
```

Comment:
we can remove the `let` here and the `response` at the bottom, returning from the `match`